app/notifications/process_notifications.py

```diff
@@ -120,6 +122,12 @@ def persist_notification(
     notification_created_at = created_at or datetime.utcnow()
     if not notification_id:
         notification_id = uuid.uuid4()
+
+    if template_has_unsubscribe_link and not unsubscribe_link:
+        base_url = current_app.config["API_HOST_NAME"]
+        url = f"/unsubscribe/{str(notification_id)}/"
+        unsubscribe_link = url_with_token(recipient, url=url, base_url=base_url)
```

Comment:
ohh, so recipient data will be hashed, but available to us later, correct?

Reply:
Encoding the recipient data is primarily to ensure the validity of an unsubscribe request. The endpoint checks that it is a valid token.
The recipient data would also be available on the `notification` object.